pdata/pcommon/common.go

```diff
@@ -772,6 +791,16 @@ func (m Map) Upsert(k string, v Value) {
 	}
 }
 
+// UpsertEmpty inserts or updates an empty value to the map under given key.
+func (m Map) UpsertEmpty(k string) {
```

Comment:
I found few places in contrib where returning a Value here will help. Reason is that for example you have a logic to modify a Value (let's say for LogRecord.Body()) and want to use the same logic here, it is impossible to do so.

See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go#L29 where I will simply append an `EmptyValue` then will initialize it in the `SetValue` call.

Reply:
Makes sense. It just fells a bit inconsistent that this method returns a wrapper - Value while all others return the child value. Maybe we call it differently? Just `Upsert` or `UpsertValue`?

Reply:
I prefer `UpsertEmpty` -> returns the empty value, since it is consistent with `UpsertEmtpyMap` -> returns the empty Map inside value

Reply:
Ok, Let's keep call it `UpsertEmpty`. I'll update the PR and will add test coverage